### PR TITLE
Anthony Catalfo

### PR DESCRIPTION
**Pull request by AnthonyCatalfo**
## Trello Set Up:
* [ ] Use your existing Trello account from the Front End Project, or create a new one
* [ ] Create a new board called "Lambda Notes(Backend) - {Your Name}"
* [ ] Create lists titled `backlog`,`To Do`, `In Progress`, and `Done`
* [ ] Fill in the `To Do` list with the MVP features listed below
* [ ] Fill in the `backlog` list with all the extra features listed below
* [ ] Share your board with the project manager that has been assigned to you.  If you have not been assigned yet, reach out to your Section Lead for guidance
* [ ] Add your Trello URL to your project's README.md file.  Commit the change, push it to your repository & submit a pull request

## Backend MVP Features:
* [x] Create a Database for your app
* [ ] Deploy your application to the web
* [x] Create a Node app and connect it to your database
* [x] Create a Registration Page that allows users to create accounts for your app and sign in with email/password
* [x] Store these User Accounts in your Database
* [x] Add a Sign In Page
* [x] Store Notes in your Database

Build an API Endpoint in your Node App for each of the following, and connect them to your React App. These should all be read from your Database and scoped to the logged in user.
* [x] Display a list of notes
* [x] Create a note with a title and content
* [x] View an existing note
* [x] Edit an existing note
* [x] Delete an existing note

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
 .DS_Store
+node_modules/
```
Comment (mixelpixel) [inline .gitignore]: Thank you for adding this to your .gitignore file ⭐️

File: README.md [hunk 2]
```diff
@@ -1,89 +1,80 @@
...
-Build an API Endpoint in your Node App for each of the following, and connect them to your React App. These should all be read from your Database and scoped to the logged in user.
-* [ ] Display a list of notes
-* [ ] Create a note with a title and content
-* [ ] View an existing note
-* [ ] Edit an existing note
-* [ ] Delete an existing note
-* [ ] Write tests for your endpoints and models
+# Pushing you Onward with your knowledge of Authentication.
+
+* **DISCLAIMER** Authentication is a subject that many people spend a large amount time throughout their careers obtaining knowledge over. This is not something we expect you to have a mastery over, rather, we're preparing you to be able have an intelligent conversation about the subject.
```
Comment (mixelpixel) [inline README.md]: huh?

File: app.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const { server } = require('./server');
+const mongoose = require('mongoose');
+const port = process.env.PORT || 5000;
+
+mongoose.Promise = global.Promise;
+mongoose.connect('mongodb://lambdauser1:lambdauser1@ds239439.mlab.com:39439/notes', {
+  useMongoClient: true
+});
```
Comment (mixelpixel) [inline app.js]: Glad to see you making use of MLab. Would be good to add error handling in case the MongoDB is not available

File: package.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "back-end-project-week",
+  "version": "1.0.0",
+  "description": "back-end-project-week",
+  "main": "server.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "nodemon app.js"
```
Comment (mixelpixel) [inline package.json]: nice - thanks for setting this start script up. Are you using the test script?

File: server.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+const cors = require('cors');
+
+const routes = require('./api/routes/routes');
+
```
Comment (mixelpixel) [inline server.js]: I like the way you've separatd your package requirements from the file requirements 👍

**Comment by mixelpixel** [comment]
Front end layout is a little funky:

screen shot 2018-04-09 at 8 12 06 pm
screen shot 2018-04-09 at 8 12 21 pm

**Comment by mixelpixel** [comment]
@AnthonyCatalfo Thanks for showing me your edit/update features with the viddy. I added the Task List to the first comment of your Pull Request and have checked off the completed items. I don't want to distract you from this week's Computer Architecture curriculum, but it would be great to see your front end deployed with the server as well as the DB you've got on Mlabs. There's a whole big wide world of Deployment, but maybe this video will help: https://youtu.be/-qrTgC15eFo

Jonathan and Luis deploy through Microsoft's Azure in the video. Other deployment options:
- Netlify: https://www.netlify.com/blog/2016/09/29/a-step-by-step-guide-deploying-on-netlify/
- GitHub Pages: https://pages.github.com/
- GH: https://help.github.com/articles/configuring-a-publishing-source-for-github-pages/
- ngrok: https://ngrok.com/
- Heroku: https://devcenter.heroku.com/articles/getting-started-with-nodejs#introduction
  - https://devcenter.heroku.com/articles/mongolab
  - https://forum.freecodecamp.org/t/guide-for-using-mongodb-and-deploying-to-heroku/19347
- MLab (for MongoDB hosting): https://mlab.com/signup/?gclid=EAIaIQobChMIrLP4p4Kc2gIVlLjACh0cFgTiEAAYASABEgJyJ_D_BwE
- Microsoft Azure: https://azure.microsoft.com/en-us/free/search/
  - portal.azure.com
- MongoDB Atlas: https://www.mongodb.com/cloud/atlas/lp/general
- Digital Ocean: https://www.digitalocean.com/
- AWS free tier: https://aws.amazon.com/free/
- Google Cloud free tier: https://cloud.google.com/free/

**Comment by mixelpixel** [comment]
I really dig ngrok - if you download it, the `ngrok` command should get added to your computer so that if you have your front-end up and running on port 3000, you can run `ngrok http 3000` and it will serve up your webpage and tell you the URL others can view it with 😎 ...like right now I am serving up your web page to the world at: https://e378b83d.ngrok.io/

**Comment by mixelpixel** [comment]
see here for a demo: https://youtu.be/6LyDznLOCMg

**Comment by AnthonyCatalfo** [comment]
Awesome!!!

**Pull request closed without merging**